extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -481,10 +499,12 @@ export class AmpStoryPage extends AMP.BaseElement {
     this.getViewport().onResize(
       debounce(this.win, () => this.onResize_(), RESIZE_TIMEOUT_MS)
     );
+
     return Promise.all([
       this.beforeVisible(),
       this.waitForMediaLayout_(),
       this.mediaPoolPromise_,
+      this.maybeInitializeParallaxFx_(),
```

Comment:
You could call this method inline right before the `Promise.all`, which would allow you to make `maybeInitializeParallaxFx_` synchronous.

Reply:
Wouldn't that call it before the page becomes visible?

Reply:
It'd be the same as now, as the method will be called when the array you're passing to Promise.all is evaluated.
Did you mean to call `maybeInitializeParallaxFx_` after the other promises resolved? In this case you could execute it in the `Promise.all().then()` callback

Reply:
Oh nevermind, I thought you meant turn the function async and `await` for  `maybeInitializeParallaxFx_`